forum/src/main/java/unipd/se18/ocrcamera/forum/viewmodels/AddPost_VM.java

```diff
@@ -0,0 +1,320 @@
+	    // Notifies the event to the View
+            notifier.onParametersSendingFailed(message);
+        }
+    };
+
+    /**
+     * Adds a post to the forum
+     * More details at: {@link RequestManager},
+     * {@link RequestManager#sendRequest(Context, ArrayList)}.
+     * @param context The reference of the activity/fragment that calls this method
+     * @param title The new post's title
+     * @param message The new post's message
+     * @param author The new post's author
+     * @author Pietro Prandini (g2)
+     */
+    @Override
+    public void addPostToForum(final Context context, String title, String message, String author) {
+        Log.i(TAG,"addPostToForum");
+
+        // Checks the validity of the parameters
+        if(!areParametersValid(title, message, author)) {
+            Log.d(TAG,"addPostToForum -> The parameters are not valid");
+
+            // The parameters are not valid
+	    // Notifies the event to the View
+            notifier.onNotValidParameters(context.getString(R.string.not_valid_parameters));
+
+            // Ends the method
+            return;
+        }
+
+        // Sets up the manager useful for adding posts
+        RequestManager postManager = new RequestManager();
+
+        // Sets up the manager worker listener
+        postManager.setOnRequestFinishedListener(requestManagerListener);
+
+        /*
+        The next try catch handles a possible failure on the creation
+        of the JSON object that represents a new post.
+         */
+
+        try {
+            // Sets up the parameters for the adding post request
+            ArrayList<RequestManager.Parameter> postManagerParameters =
+                    getAddPostParameters(title, message, author);
+
+            // Sends the request
+            Log.i(TAG,"addPostToForum -> Sending the request");
+            postManager.sendRequest(context, postManagerParameters);
+        } catch (JSONException e) {
+            // JSON Post creation failed
+            e.printStackTrace();
+            // Notifies the error
+            Log.d(TAG, "addPostToForum -> JSON creation problem");
+	    // Notifies the event to the View
+            notifier.onJSONPostCreationFailed(e.toString());
+        }
+    }
+
+    /**
+     * Checks the validity of the parameters received (not null and not empty String)
+     * @param title The new post's title
+     * @param message The new post's message
+     * @param author The new post's author
+     * @return TRUE if the parameters are valid (not null and not empty String), FALSE otherwise
+     * @author Pietro Prandini (g2)
+     */
+    private Boolean areParametersValid(String title, String message, String author) {
+        // Checks if the Strings are not null and not empty ones
+        return (title != null && !title.equals(""))
+                && (message != null && !message.equals(""))
+                && (author != null && !author.equals(""));
```

Comment:
I've to document about that class because it has some interesting method and it could be useful here or anyway in the future, so thanks you for sharing this information.

Reply:
I'm glad to hear that!